lib/components/admin/mailables-window.js

```diff
@@ -0,0 +1,194 @@
+    if (!callTaker.mailables.visible) return null
+    const selectableMailables = items.filter(m =>
+      !selectedMailables.find(mailable => mailable.name === m.name)
+    )
+    return (
+      <DraggableWindow
+        header={
+          <>
+            <WindowHeader>
+              <Icon type='graduation-cap' /> Mailables{' '}
+              <span className='pull-right'>
+                <button
+                  className='clear-button-formatting'
+                  onClick={this._onClickCreateLetter}
+                  style={{marginRight: '5px', verticalAlign: 'bottom'}}
+                >
+                  Create Letter
+                </button>
+              </span>
+            </WindowHeader>
+          </>
+        }
+        onClickClose={toggleMailables}
+        scroll={false}
+        style={{height: '340px', width: '600px'}}
+      >
+        <div>
+          {MAILABLE_FIELDS.map(f => (
+            <input
+              key={f.fieldName}
+              id={f.fieldName}
+              onChange={this._updateField}
+              placeholder={f.placeholder}
+              style={{margin: '5px'}}
+              value={this.state[f.fieldName]} />
+          ))}
+        </div>
+        <div style={{display: 'flex'}}>
+          <div style={{width: '300px'}}>
+            <h4>All Mailables</h4>
+            <MailablesList>
+              {selectableMailables.map((mailable, i) => (
+                <MailableOption
+                  key={mailable.name}
+                  mailable={mailable}
+                  onClick={this._addMailable} />
+              ))}
+            </MailablesList>
+          </div>
+          <div style={{width: '300px'}}>
+            <h4>Selected Mailables <Badge>{selectedMailables.length}</Badge></h4>
+            <MailablesList>
+              {selectedMailables.length > 0
+                ? selectedMailables.map((mailable, i) => (
+                  <MailableOption
+                    index={i}
+                    key={mailable.name}
+                    mailable={mailable}
+                    onClear={this._removeMailable}
+                    updateField={this._updateMailableField} />
+                ))
+                : <div className='text-muted'>No mailables selected.</div>
+              }
+            </MailablesList>
+          </div>
+        </div>
+      </DraggableWindow>
+    )
+  }
+}
+
+class MailableOption extends Component {
+  _changeLargeFormat = (evt) => {
+    const {index, updateField} = this.props
+    updateField(index, 'largeFormat', evt.target.checked)
+  }
+
+  _changeQuantity = (evt) => {
+    const {index, updateField} = this.props
+    updateField(index, 'quantity', evt.target.value)
+  }
+
+  _onClear = () => this.props.onClear && this.props.onClear(this.props.mailable)
+
+  _onClick = () => this.props.onClick && this.props.onClick(this.props.mailable)
+
+  render () {
+    const {mailable, onClear} = this.props
+    const isSelected = Boolean(onClear)
```

Comment:
It might be more straightforward to have `isSelected` be a boolean prop instead of deriving it like this.

Reply:
Actually, scratch that, this component should actually be two components (`SelectedMailable` and (`SelectableMailable` or `MailableOption`)) so that the [Single Responsibility Principle](https://deviq.com/principles/single-responsibility-principle) can be adhered to.